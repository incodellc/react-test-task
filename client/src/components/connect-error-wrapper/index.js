import ConnectErrorWrapper from './ConnectErrorWrapper';
export default ConnectErrorWrapper;