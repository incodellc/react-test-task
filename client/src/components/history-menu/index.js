import HistoryMenu from './HistoryMenu';
export default HistoryMenu;