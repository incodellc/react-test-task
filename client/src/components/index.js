import InfoBar from './info-bar';
import Layout from './layout';
import ConnectErrorWrapper from './connect-error-wrapper';
import TickerList from './ticker-list';
import TickerListItem from './ticker-list-item';

export { InfoBar, Layout, ConnectErrorWrapper, TickerList, TickerListItem };