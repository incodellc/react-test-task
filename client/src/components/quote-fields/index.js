import QuoteFields from './QuoteFields';
export default QuoteFields;