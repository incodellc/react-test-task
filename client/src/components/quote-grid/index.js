import QuotesGrid from './QuotesGrid';
export default QuotesGrid;