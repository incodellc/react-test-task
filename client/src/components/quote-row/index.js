import QuoteRow from './QuoteRow';
export default QuoteRow;