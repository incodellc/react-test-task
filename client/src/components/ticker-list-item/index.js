import TickerListItem from './TickerListItem';
export default TickerListItem;