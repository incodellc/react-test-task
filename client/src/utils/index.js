export const tickers = [
    {
        tickerId: 'AAPL',
        company: 'Apple Inc.'
    },
    {
        tickerId: 'GOOGL',
        company: 'Alphabet Inc.'
    },
    {
        tickerId: 'AMZN',
        company: 'Amazon.com, Inc.'
    },
    {
        tickerId: 'ADBE',
        company: 'Adobe Systems Inc.'
    },
    {
        tickerId: 'CSCO',
        company: 'Cisco Systems, Inc.'
    },
    {
        tickerId: 'DISCK',
        company: 'Discovery Inc.'
    },
    {
        tickerId: 'EBAY',
        company: 'eBay Inc.'
    },
    {
        tickerId: 'NFLX',
        company: 'Netflix'
    },
    {
        tickerId: 'MSFT',
        company: 'Microsoft Corporation'
    },
    {
        tickerId: 'TSLA',
        company: 'Tesla Motors, Inc.'
    }
];

export const fields = [
    {
        name: 'exchanger',
        label: 'Exchange',
        isDate: false,
        sortable: false
    },
    {
        name: 'ticker',
        label: 'Ticker',
        isDate: false,
        sortable: true
    },
    {
        name: 'price',
        label: 'Price',
        isDate: false,
        sortable: true
    },
    {
        name: 'change',
        label: 'Change',
        isDate: false,
        sortable: true
    },
    {
        name: 'change_percent',
        label: 'Change %',
        isDate: false,
        sortable: true
    },
    {
        name: 'dividend',
        label: 'Dividend',
        isDate: false,
        sortable: true
    },
    {
        name: 'yield',
        label: 'Yield',
        isDate: false,
        sortable: true
    },
    {
        name: 'last_trade_time',
        label: 'Last trade time',
        isDate: true,
        sortable: true
    }
];

export const rewriteObjectProps = (prevObject, newPropsValues) => {
    return {
        ...prevObject,
        ...newPropsValues
    };
};