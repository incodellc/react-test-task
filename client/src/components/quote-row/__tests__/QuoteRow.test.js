import React from 'react';
import { fields, quotesExample } from '../../../utils';
import { renderWithRedux } from '../../../utils/test-utils';
import { QuoteRow } from '../../../components';
import { Table } from 'semantic-ui-react';
import { cleanup } from '@testing-library/react';
import moment from 'moment';

describe('Check QuoteRow component', () => {
    afterEach(cleanup);

    it('component renders correctly', () => {
        const { asFragment } = renderWithRedux(<Table><Table.Body><QuoteRow data={quotesExample[0]} /></Table.Body></Table>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('component should render row of cells according to order and number of given fields', () => {
        const { getByTestId, getAllByTestId } = renderWithRedux(<Table><Table.Body><QuoteRow data={quotesExample[0]} /></Table.Body></Table>);
        const cells = getAllByTestId('quote-cell');
        expect(getByTestId('quote-row')).toContainElement(cells[0]);
        expect(cells.length).toEqual(fields.length);
        fields.forEach((field, index) => {
            expect(cells[index]).toHaveTextContent(field.isDate ? moment(quotesExample[0][field.name]).format('DD-MM-YY HH:mm:ss') : quotesExample[0][field.name]);
        });
    });

    it('component should highlight negative changes of price', () => {
        const { getByTestId } = renderWithRedux(<Table><Table.Body><QuoteRow data={quotesExample[0]} /></Table.Body></Table>);
        expect(getByTestId('quote-row')).toHaveClass('negative');
        expect(getByTestId('quote-row')).toContainElement(getByTestId('quote-cell-icon-down'));
        expect(getByTestId('quote-cell-icon-down')).toHaveClass('arrow alternate circle down');
    });
    
    it('component should highlight positive changes of price', () => {
        const { getByTestId } = renderWithRedux(<Table><Table.Body><QuoteRow data={quotesExample[1]} /></Table.Body></Table>);
        expect(getByTestId('quote-row')).toHaveClass('positive');
        expect(getByTestId('quote-row')).toContainElement(getByTestId('quote-cell-icon-up'));
        expect(getByTestId('quote-cell-icon-up')).toHaveClass('arrow alternate circle up');
    });
});