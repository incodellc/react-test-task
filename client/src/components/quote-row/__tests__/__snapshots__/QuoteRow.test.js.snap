// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check QuoteRow component component renders correctly 1`] = `
<DocumentFragment>
  <table
    class="ui table"
  >
    <tbody
      class=""
    >
      <tr
        class="negative"
        data-testid="quote-row"
      >
        <td
          class=""
          data-testid="quote-cell"
        >
          NASDAQ
        </td>
        <td
          class=""
          data-testid="quote-cell"
        >
          AAPL
        </td>
        <td
          class=""
          data-testid="quote-cell"
        >
          <i
            aria-hidden="true"
            class="arrow alternate circle down icon"
            data-testid="quote-cell-icon-down"
          />
          213.41
        </td>
        <td
          class=""
          data-testid="quote-cell"
        >
          -38.72
        </td>
        <td
          class=""
          data-testid="quote-cell"
        >
          -0.18
        </td>
        <td
          class=""
          data-testid="quote-cell"
        >
          0.56
        </td>
        <td
          class=""
          data-testid="quote-cell"
        >
          0.55
        </td>
        <td
          class=""
          data-testid="quote-cell"
        >
          15-02-20 23:40:17
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
